posthog/hogql/database/schema/util/session_where_clause_extractor.py

```diff
@@ -0,0 +1,329 @@
+                        ),
+                    ]
+                )
+            elif node.op == CompareOperationOp.Gt or node.op == CompareOperationOp.GtEq:
+                return ast.CompareOperation(
+                    op=ast.CompareOperationOp.GtEq,
+                    left=ast.ArithmeticOperation(
+                        op=ast.ArithmeticOperationOp.Add,
+                        left=rewrite_timestamp_field(node.left),
+                        right=ast.Call(name="toIntervalDay", args=[ast.Constant(value=SESSION_BUFFER_DAYS)]),
+                    ),
+                    right=node.right,
+                )
+            elif node.op == CompareOperationOp.Lt or node.op == CompareOperationOp.LtEq:
+                return ast.CompareOperation(
+                    op=ast.CompareOperationOp.LtEq,
+                    left=ast.ArithmeticOperation(
+                        op=ast.ArithmeticOperationOp.Sub,
+                        left=rewrite_timestamp_field(node.left),
+                        right=ast.Call(name="toIntervalDay", args=[ast.Constant(value=SESSION_BUFFER_DAYS)]),
+                    ),
+                    right=node.right,
+                )
+        elif is_right_timestamp_field and is_left_constant:
+            # let's not duplicate the logic above, instead just flip and it and recurse
+            if node.op in [
+                CompareOperationOp.Eq,
+                CompareOperationOp.Lt,
+                CompareOperationOp.LtEq,
+                CompareOperationOp.Gt,
+                CompareOperationOp.GtEq,
+            ]:
+                return self.visit(
+                    ast.CompareOperation(
+                        op=CompareOperationOp.Eq
+                        if node.op == CompareOperationOp.Eq
+                        else CompareOperationOp.Lt
+                        if node.op == CompareOperationOp.Gt
+                        else CompareOperationOp.LtEq
+                        if node.op == CompareOperationOp.GtEq
+                        else CompareOperationOp.Gt
+                        if node.op == CompareOperationOp.Lt
+                        else CompareOperationOp.GtEq,
+                        left=node.right,
+                        right=node.left,
+                    )
+                )
+
+        return ast.Constant(value=True)
+
+    def visit_arithmetic_operation(self, node: ast.ArithmeticOperation) -> ast.Expr:
+        # don't even try to handle complex logic
+        return ast.Constant(value=True)
+
+    def visit_not(self, node: ast.Not) -> ast.Expr:
+        return ast.Constant(value=True)
+
+    def visit_call(self, node: ast.Call) -> ast.Expr:
+        if node.name == "and":
+            return self.visit_and(ast.And(exprs=node.args))
+        elif node.name == "or":
+            return self.visit_or(ast.Or(exprs=node.args))
+        return ast.Constant(value=True)
+
+    def visit_field(self, node: ast.Field) -> ast.Expr:
+        return ast.Constant(value=True)
```

Comment:
I'm not so sure about some of these, turning a `ast.Field` into a constant may have some side effects. What happens if we write a subquery in a where clause? I think that would totally break, maybe

Reply:
I just added a comment that explains how this is working, but essentially it returns True for any expr it can't reason about, always including more rows than are necessary